mitodl/edx-platform
# Restored legacy dashboard enrollment functionality

#### What are the relevant tickets?

Closes mitocw#76
Closes #12 

#### What's this PR do?

- Gives instructors the ability to see what students are enrolled in specific sections of a course in the remote gradebook (was removed after the dogwood releases)
- Changes 'grade export' naming convention for files/functions/variables/etc. to 'remote gradebook' since the tab includes functionality that isn't directly related to grade exporting
-
- Several UI improvements, including a select box for section names & assignment names that populates via API call (instead of echoing possible values on a button click and requiring users to copy one of those values into a text input)
- Refactors several API methods for correctness, performance, and cleanliness

#### Where should the reviewer start?

Probably `lms/templates/instructor/instructor_dashboard_2/remote_gradebook.html` and  `lms/djangoapps/instructor/views/api.py`

#### How should this be manually tested?

- Set the `ENABLE_INSTRUCTOR_REMOTE_GRADEBOOK_CONTROLS ` feature to `True`
- Set a `REMOTE_GRADEBOOK_URL` feature value - you can ask me or Peter for the proper setting here
- Navigate to the `Remote Gradebook` tab for some course (eg: `http://localhost:8000/courses/course-v1:edX+DemoX+Demo_Course/instructor#view-remote_gradebook` - make sure you have staff/admin privileges)
- Choose a section and click 'List students in section in remote gradebook' to make sure it works
- Click the other buttons and make sure they still work

#### Any background context you want to provide?

Keep in mind that a few files were renamed when reviewing the diff!
